### PR TITLE
SAK-43112 Recover rubrics with softly deleted assignments

### PR DESCRIPTION
**Pull request by bgarciaentornos**
NOTE: Parameters order on deleteRubricAssociation method was wrong in the service declaration.

### CODE REVIEW

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 9
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=4922915&bid=16289151)

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 1]
```diff
@@ -9412,7 +9412,7 @@ public void doDelete_assignment(RunData data) {
                 String title = assignment.getTitle();
 
                 // remove rubric association if there is one
-                rubricsService.deleteRubricAssociation(RubricsConstants.RBCS_TOOL_ASSIGNMENT, id);
+                rubricsService.softDeleteRubricAssociation(RubricsConstants.RBCS_TOOL_ASSIGNMENT, id);
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: feel like we don't need to softly delete a rubric association at all?

Comment (bgarciaentornos) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: This is done to unlock the rubric and allow removing it, keeping consistency with the approach taken on https://github.com/sakaiproject/sakai/pull/7749

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=4922915&bid=16289151)

**Comment by ern** [comment]
Going to merge this PR as it makes it consistent with the pattern that was established in #7749 but I still have reservations about this approach.

**Pull request merged**